.github/workflows/misc-linters.yml

```diff
@@ -33,3 +33,44 @@ jobs:
           pip install codespell
           # exclude files which may be synchronized from other places
           git ls-files | grep -v "^thirdparty" | grep -v "/thirdparty/" | grep -v "/dist/" | xargs -t codespell --ignore-words .github/workflows/ignore_words 2>/dev/null
+
+  linelint:
+    name: linelint # Explicit job name
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read # Restrict permissions for security
+    steps:
+      - name: Harden the runner (Audit all outbound calls)
+        uses: step-security/harden-runner@6c439dc8bdf85cadbbce9ed30d1c7b959517bc49 # v2.12.2
+        with:
+          egress-policy: audit
+      - name: Checkout code
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          fetch-depth: 0 # Needed for git diff
+      - name: Ensure git safe directory
+        run: |
+          git config --global --add safe.directory $(pwd)
+      - name: Get changed files
+        id: changed-files
+        run: |
+          BASE_REF=${{ github.base_ref }}
+          if git rev-parse --verify origin/$BASE_REF >/dev/null 2>&1; then
+            FILES=$(git diff --name-only --diff-filter=d origin/$BASE_REF...HEAD | grep -v "^thirdparty/" | grep -v "/dist/" | tr '\n' ' ')
+          else
+            FILES=$(git diff --name-only --diff-filter=d HEAD~1...HEAD | grep -v "^thirdparty/" | grep -v "/dist/" | tr '\n' ' ')
+          fi
+          if [ -z "$FILES" ]; then
+            echo "No files changed, skipping linelint."
+            exit 0
+          fi
+          echo "files=$FILES" >> $GITHUB_OUTPUT
```

Comment:
Could you please explain the logic you are using here? For example, why does the `else` condition only select `HEAD~1`?